src/init.c

```diff
@@ -67,6 +67,21 @@ static char shmem_internal_my_hostname[MAXHOSTNAMELEN];
 static char shmem_internal_my_hostname[HOST_NAME_MAX];
 #endif
 
+
+static inline long errchk_atol(char *s)
+{
+    long val;
+    char *e;
+    errno = 0;
+
+    val = strtol(s,&e,0);
+    if(errno != 0) {
+        perror("env var conversion");
+        exit(1);
```

Comment:
If the runtime is already initialized, we should use shmem_runtime_abort(...) here.